### PR TITLE
Es/create or join project

### PR DESCRIPTION
**Pull request by ErikSin**
image

image

image

### CODE REVIEW

File: src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  const {
+    control,
+    handleSubmit,
+    formState: {errors},
+    watch,
+  } = useForm<ProjectFormType>({defaultValues: {projectName: ''}});
+
+  function handleCreateProject(val: ProjectFormType) {
+    mutate(val.projectName);
+    navigation.navigate('ProjectCreated', {name: val.projectName});
```
Comment (achou11) [inline src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx:60]: this shouldn't navigate unless the mutation succeeds. was able to update the mutation implementation to throw an error and the navigation occurred anyways (and the project wasn't created despite the UI saying it was).

not sure what the "idiomatic" react query way of updating this is, but seems like [`mutateAsync`](https://tanstack.com/query/latest/docs/react/guides/mutations#promises) might be what's needed here?

```ts
try {
  await mutateAsync(val.projectName)
  navigation.navigate('ProjectCreated', {name: val.projectName});
} catch (err) {
  // Failed to create project, surface this error somewhere in the UI
}
```

**Review by achou11 — CHANGES_REQUESTED** [review]
Noticed a couple of issues of notable issues, but overall the changes seem good!

File: src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx (hunk 1 above)
Comment (ErikSin) [reply src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx:60]: `mutate` actually as a [callback that you can use directly](https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync)

Comment (ErikSin) [reply src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx:60]: We need to figure out what we are doing with our errors....

Comment (ErikSin) [reply src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx]: https://github.com/digidem/CoMapeo-mobile/pull/77/commits/a30305e93468aedff6c33f3192648d423dff529b

File: src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+              style={styles.accordianHeader}>
+              <Text>{t(m.advancedSettings)}</Text>
+              <MaterialIcon
+                name={
+                  !advancedSettingOpen
+                    ? 'keyboard-arrow-up'
+                    : 'keyboard-arrow-down'
+                }
+                size={40}
+              />
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx]: https://github.com/digidem/CoMapeo-mobile/pull/77/commits/a30305e93468aedff6c33f3192648d423dff529b

**Comment by ErikSin** [comment]
closing to refactor using error handling and react hook forms

**Pull request closed without merging**